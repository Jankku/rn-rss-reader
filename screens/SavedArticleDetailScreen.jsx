import { useEffect, useState } from 'react';
import { View } from 'react-native';
import WebView from 'react-native-webview';
import ArticleSaveButton from '../components/NewsDetail/ArticleSaveButton';
import SavedArticleController from '../data/local/SavedArticleController';
import useToast from '../utils/useToast';

function SavedArticleDetailScreen({ navigation, route }) {
  const guid = route.params.guid;
  const [article, setArticle] = useState();
  const [isSaved, setIsSaved] = useState();
  const { showToast } = useToast();

  useEffect(() => {
    const item = SavedArticleController.getArticleById(guid);
    setArticle(item);

    const saved = SavedArticleController.isArticleSaved(guid);
    setIsSaved(saved);
  }, [guid]);

  useEffect(() => {
    if (article) {
      navigation.setOptions({
        title: article.title,
        headerTitleAlign: 'left',
        headerRight: () => (
          <ArticleSaveButton
            isSaved={isSaved}
            onPress={() => {
              if (article && isSaved) deleteArticleAction();
            }}
          />
        ),
      });
    }
  }, [article, isSaved]);

  const deleteArticleAction = () => {
    try {
      SavedArticleController.deleteArticle(article);
      setIsSaved(false);
      showToast('Article deleted');
      navigation.goBack();
    } catch (error) {
      showToast("Error: Couldn't delete article");
    }
  };

  return (
    <View style={{ flex: 1 }}>
      {article ? (
        <WebView
          source={{
            uri: article.link,
          }}
        />
      ) : null}
    </View>
  );
}

export default SavedArticleDetailScreen;
