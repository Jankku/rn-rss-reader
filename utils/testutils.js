import { render } from '@testing-library/react-native';
import { ThemeContextProvider } from '../context/ThemeContext';
import { LocationContextProvider } from '../context/LocationContext';
import { RegionContextProvider } from '../context/RegionContext';

const AllProviders = ({ children }) => {
  return (
    <LocationContextProvider>
      <RegionContextProvider>
        <ThemeContextProvider>{children}</ThemeContextProvider>
      </RegionContextProvider>
    </LocationContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
