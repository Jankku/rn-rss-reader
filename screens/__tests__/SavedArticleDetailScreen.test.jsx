import SavedArticleDetailScreen from '../SavedArticleDetailScreen';
import { render } from 'testutils';

describe('Test SavedArticleDetailScreen', () => {
  test('Should match snapshot', () => {
    const { toJSON } = render(
      <SavedArticleDetailScreen
        navigation={{ navigate: () => {}, setOptions: () => {} }}
        route={{ params: { guid: 'https://yle.fi/uutiset/3-12345678' } }}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
