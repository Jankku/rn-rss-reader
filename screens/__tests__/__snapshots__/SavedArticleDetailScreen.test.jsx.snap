// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SavedArticleDetailScreen Should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
